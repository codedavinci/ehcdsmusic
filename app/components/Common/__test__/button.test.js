import React from 'react'

import { Button } from '../'

describe('<Button />', () => {
  global.console = { error: jest.fn() }

  const doShallow = (overrides = {}) => {
    const button = shallow(<Button {...overrides}>Submit</Button>)
    return button
  }

  it('renders', () => {
    const button = doShallow()
    expect(button).toMatchSnapshot()
  })

  it('should have the size small as default', () => {
    let button = doShallow()
    expect(button).toHaveClassName('custom-button', 'small')
  })

  it('should not be allowed to render without children', () => {
    shallow(<Button />)
    expect(console.error).toHaveBeenCalled()
  })

  it('should be able to inject new props or attributes', () => {
    let button = doShallow({ name: 'btn' })
    expect(button).toHaveProp('name', 'btn')
  })

  it('should be able to disable', () => {
    let button = doShallow({ disabled: true })
    expect(button).toHaveProp('disabled')
  })
})
