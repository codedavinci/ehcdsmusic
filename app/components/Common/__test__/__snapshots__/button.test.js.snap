// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders 1`] = `
<button
  className="custom-button small"
>
  Submit
</button>
`;
